test/models/rpm_test.rb

```diff
@@ -44,38 +46,52 @@ def test_in_repositories_uniqness
       assert_equal Rpm.in_repositories([@repo, repo2]).to_a.sort, [@rpm_one, @rpm_two].sort
     end
 
-    def test_with_search
+    def test_search_version_greater_than_or_equal
       rpms = Rpm.in_repositories(@repo).non_modular.search_for('version >= 1.0')
-      expected = [@rpm_one, @rpm_one_two, @rpm_three, @rpm_two]
-      assert_equal expected, rpms.to_a.sort
+      expected = [@rpm_one, @rpm_one_i686, @rpm_one_i686_two, @rpm_one_two, @rpm_three, @rpm_two]
+      assert_equal expected.sort_by(&:id), rpms.to_a.sort_by(&:id)
```

Comment:
RPMs are sorted by name, but all the RPMs here have the same name. So, we need to sort by ID.